### DISCUSSION TITLE
RemoteFX USB Passthrough for HID devices - workarounds?

### DISCUSSION THREAD

**Discussion by victorhooi** [FreeRDP/FreeRDP#8144] [Q&A]
_(This is a follow-up from [here](https://github.com/FreeRDP/FreeRDP/discussions/7635), focusing on the HID part)_

I have a Windows Server 2022 instance I'm trying to connect to, using FreeRDP.

I would like to pass through a Yubikey for U2F functionality in the Chrome browser, running

My understanding is that the Yubikey U2F functionality is exposed as a USB HID device.

There is a blog post [here](https://queensidecastle.com/guides/use-a-yubikey-remotely-over-rdp) that talks about passing through Yubikey U2F via RDP - apparently you need to do Windows Group Policy changes on both the server, and the client. The client they are using is the normal Microsoft Windows MSTC client.

I've already done the server side Group Policy changes:

![image](https://user-images.githubusercontent.com/243499/185757953-64ce329d-af75-4715-8a8f-96fc4c6e3727.png)

For the client side - the [blog post](https://queensidecastle.com/guides/use-a-yubikey-remotely-over-rdp) talks about enabling RemoteFX USB redirection, and also adding the GUID entries for the Yubikey to the list of allowable USB devices.

Does anybody know what the equivalent to these changes are for using FreeRDP as a client?

Or what else do you need to passthrough USB HID devices successfully?

(NB: I know in the [earlier discussion](https://github.com/FreeRDP/FreeRDP/discussions/7635#discussioncomment-3186448) mfleisz@ mentioned that HID redirection isn't officially supported, but nothing specifically stops it - it wasn't clear though if you needed to do something to enable it?)

**Comment by mfleisz** [comment]
Hi @victorhooi 

all you should have to do is specifying the device you want to redirect using the `/usb` switch i.e. `/usb:id:054c:0268`

**Comment by hardening** [comment]
Note: exporting the yubikey with smartcard redirection works, even if I think that then the yubikey manager doesn't work as it relies on the USB interface of the key. May worth give a try...
